math/overflow.go

```diff
@@ -1,11 +1,94 @@
 package math
 
-// AddUint64 performs addition on uint64 and returns an error if the addition overflows
-func AddUint64(a, b uint64) (uint64, error) {
+import (
+	builtinMath "math"
+
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+)
+
+var log = logger.GetOrCreate("arwen/overflow")
+
+// AddUint64 performs addition on uint64 and logs an error if the addition overflows
+func AddUint64(a, b uint64) uint64 {
+	res, err := AddUint64WithErr(a, b)
+	if err != nil {
+		log.Error("AddUint64 overflow", a, b)
+		return builtinMath.MaxUint64
+	}
+
+	return res
+}
+
+// AddUint64WithErr performs addition on uint64 and returns an error if the addition overflows
+func AddUint64WithErr(a, b uint64) (uint64, error) {
 	s := a + b
 	if s >= a && s >= b {
 		return s, nil
 	}
 
 	return s, ErrAdditionOverflow
 }
+
+// AddInt64 performs addition on int64 and logs an error if the addition overflows
+func AddInt64(a, b int64) int64 {
+	res := a + b
+	if (res > a) == (b > 0) {
+		return res
+	}
+
+	log.Error("AddInt64 overflow", a, b)
+	return builtinMath.MaxInt64
+}
+
+// SubUint64 performs subtraction on uint64 and logs an error if the subtraction overflows
+func SubUint64(a, b uint64) uint64 {
```

Comment:
maybe keep the same standard as with Add - where you have SubUnitWithError as well.

Reply:
Done.